lib/unittest.bzl

```diff
@@ -196,22 +273,27 @@ def _end(env):
     that the results are reported.
 
     Args:
-      env: The test environment returned by `unittest.begin`.
+      env: The test environment returned by `unittest.begin` or `analysistest.begin`.
     """
-
-    tc = env.ctx.toolchains[TOOLCHAIN_TYPE].unittest_toolchain_info
-    testbin = env.ctx.actions.declare_file(env.ctx.label.name + tc.file_ext)
-    if env.failures:
-        cmd = tc.failure_templ % tc.join_on.join(env.failures)
+    if getattr(env.ctx.attr, _IS_ANALYSIS_TEST_ATTR_NAME, False):
+        return [AnalysisTestResultInfo(
+            success = (len(env.failures) == 0),
+            message = "\n".join(env.failures),
+        )]
     else:
```

Comment:
Given the above returns, maybe drop the `else:` and drop the indent on the rest of this?